lib/web/integrations_awsoidc.go

```diff
@@ -866,32 +867,52 @@ func (h *Handler) awsOIDCConfigureIdP(w http.ResponseWriter, r *http.Request, p
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
-	s3Bucket := queryParams.Get("s3Bucket")
-	s3Prefix := queryParams.Get("s3Prefix")
-	if s3Bucket == "" || s3Prefix == "" {
-		return nil, trace.BadParameter("s3Bucket and s3Prefix query params are required")
-	}
-	s3URI := url.URL{Scheme: "s3", Host: s3Bucket, Path: s3Prefix}
-
-	jwksContents, err := h.jwks(r.Context(), types.OIDCIdPCA)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	jwksJSON, err := json.Marshal(jwksContents)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
 	// The script must execute the following command:
 	// teleport integration configure awsoidc-idp
 	argsList := []string{
 		"integration", "configure", "awsoidc-idp",
 		fmt.Sprintf("--cluster=%s", shsprintf.EscapeDefaultContext(clusterName)),
 		fmt.Sprintf("--name=%s", shsprintf.EscapeDefaultContext(integrationName)),
 		fmt.Sprintf("--role=%s", shsprintf.EscapeDefaultContext(role)),
-		fmt.Sprintf("--s3-bucket-uri=%s", shsprintf.EscapeDefaultContext(s3URI.String())),
-		fmt.Sprintf("--s3-jwks-base64=%s", base64.StdEncoding.EncodeToString(jwksJSON)),
 	}
+
+	// We have two set up modes:
+	// - use the Proxy HTTP endpoint as Identity Provider
+	// - use an S3 Bucket for storing the public keys
```

Comment:
We always serve the oidc configuration from the proxy even if the S3 bucket is enabled, right? This comment kinda makes it seem like it's only one of the two at a time.